src/Filesystem/IsFile.php

```diff
@@ -84,10 +84,13 @@ public function asset(array|string|null $props = null): File
 			return $this->asset;
 		}
 
-		$props = $props ?? [
-			'root' => $this->root(),
-			'url'  => $this->url()
-		];
+		if (is_array($props) === false) {
+			$props = [
+				'root' => $props ?? $this->root()
+			];
+		}
```

Comment:
If we already add the model, maybe this all should rather read now

```php
$props ??= ['model' => $this];
```

Reply:
But if `$props` is not `null`, the `model` won't added. The `model` should always be added.

Reply:
Ahh, I get it now what you mean 👍

Reply:
I couldn't simplify any further. The `string` $root parameter is now useless. Should we deprecate?

````php
public function asset(array|string|null $props = null): File
{
	if ($this->asset !== null) {
		return $this->asset;
	}

	$props ??= [];

	if (is_string($props) === true) {
		$props = ['root' => $props];
	}

	$props['model'] ??= $this;

	return $this->asset = match ($this->type()) {
		'image' => new Image($props),
		default => new File($props)
	};
}
````

Reply:
> The string $props (to use as root) parameter is now useless. 

Why? You still have `is_string($props) === true`. I think this looks good.